repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -474,24 +474,20 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
           mapper.getFactory.configure(JsonGenerator.Feature.ESCAPE_NON_ASCII, true)
           Option(new ByteArrayInputStream(mapper.writeValueAsBytes(destination)))
         } catch {
-          case jpe: JsonProcessingException =>
-            throw SamlPolicyException(SC_BAD_GATEWAY, "Origin service provided bad response", jpe)
+          case e@(_: AttributeMapperException | _: JsonProcessingException) =>
```

Comment:
So this is one change where the exceptions being handled don't quite map one-to-one. I considered changing this to:
```
e@(CausedBy(_: JsonProcessingException) | _: JsonProcessingException)
```

I chose not to because the XML block below catches `SAXException` and `SaxonApiException`, both of which would come from a failed translation. Those were not being caught in this (the JSON) block, so with the updates, I made the two blocks catch the same stuff.

As for the `JsonProcessingException`, that is because we are using a function of `AttributeMapper` that was not covered by the exception handling update. Since we are using that function to parse the JSON into the appropriate data structure, we still need to handle that exception.

Reply:
I thought you were going to update those other calls? I'm guessing you didn't.

Reply:
I did, actually. The reason we are still catching the `JsonProcessingException` here is that we don't only use the `AttributeMapper` call. We also call `ObjectMapper.writeValueAsBytes` on our own object, which can also throw that exception.